framework/include/interfaces/ElementIDInterface.h

```diff
@@ -126,6 +126,9 @@ class ElementIDInterface
    */
   dof_id_type getElementID(const Elem * elem, unsigned int elem_id_index) const
   {
+    if (elem_id_index == elem->n_extra_integers())
```

Comment:
I know why you are doing this after reading the documentation and with your use case, but you really need a comment here because otherwise this is not clear at all.

Reply:
For such a simple function, we do not need excessive docs. It will take hours to make the change...